include/dragon/ast/variable_declaration_node.h

```diff
@@ -0,0 +1,23 @@
+#pragma once 
+
+#include <memory>
+#include "../ast.h"
+
+class VariableDeclarationNode : public StatementNode {
+public:
+    Token identifier;
```

Comment:
This nodes has an identifier and type attributes but could use an identifier node instead.